service/src/main/kotlin/io/provenance/digitalcurrency/consortium/frameworks/RedemptionCompletionQueue.kt

```diff
@@ -46,17 +48,27 @@ class CoinRedemptionQueue(
 
     override fun processMessage(message: CoinRedemptionDirective): CoinRedemptionOutcome {
         transaction {
-            CoinRedemptionRecord.findForUpdate(message.id).first().let { coinRedemption ->
+            CoinRedemptionRecord.findPendingForUpdate(message.id).first().let { coinRedemption ->
                 withMdc(*coinRedemption.mdc()) {
-                    check(
-                        coinRedemption.status == CoinRedemptionStatus.INSERTED ||
-                            coinRedemption.status == CoinRedemptionStatus.PENDING_REDEEM
-                    ) { "Invalid coin redemption status for queue processing" }
+                    val response = pbcService.getTransaction(coinRedemption.txHash!!)!!.txResponse
 
-                    when (coinRedemption.status) {
-                        CoinRedemptionStatus.INSERTED -> coinRedemptionService.createEvent(coinRedemption)
-                        CoinRedemptionStatus.PENDING_REDEEM -> coinRedemptionService.eventComplete(coinRedemption)
-                        else -> log.error("Invalid status - should never get here")
+                    when (response == null || response.isFailed()) {
+                        true -> {
+                            log.info("redeem failed, need to retry")
+                            coinRedemption.resetForRetry()
+                        }
+                        false -> {
+                            log.info("Completing redemption, setting up the burn")
+                            try {
+                                CoinBurnRecord.insert(
```

Comment:
with the `txResponse.logsList.first().findWithdrawEvent(` changes earlier gotta make sure batched tx's probably still work with that setup :feelsbad:

Reply:
there isn't a straightforward resolution to this with batching. Would like to discuss use cases with you tomorrow.

Reply:
might not be worth even implementing in batched world given we probably want to fully separate redemption from burning (let the bank middleware decide when to burn)